atlasdb-dagger/src/main/java/com/palantir/atlasdb/services/TransactionManagerModule.java

```diff
@@ -121,7 +122,7 @@ public SerializableTransactionManager provideTransactionManager(
         // todo(gmaretic): should this be using a real sweep queue?
         return new SerializableTransactionManager(
                 metricsManager,
-                kvs,
+                new DefaultTransactionKeyValueServiceManager(kvs),
```

Comment:
This is what I mean by "there's so many ways to construct everything". I had to again update a large amount of places, it feels like all of this should go through some shared construction logic.

Reply:
My point is a bit broader: there's just sooo many invocations of SerializableTransactionManager itself in in the majority of cases you the actual construction logic is the same. I wonder if these should be re-written to use some sort of builder and only override the pieces they need, instead of having to always construct everything.